### PR TITLE
[dag] Smoke tests and related fixes

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR copies the existing consensus smoke tests and adapts it to DAG consensus. It also includes two fixes identified from running the tests.
- Any fetch was fetching from the lowest incomplete round, but that is not realistic as validators fail to propose and rounds can be incomplete. Using the lowest incomplete round as the start round for fetch resulted in fetching >30 (3*window) rounds and failing because thats the GC limit. However, we only need to fetch the last 10 (window) rounds as that's the window we order.
- When a validator processes a certified node after its missing parents have been fetched, all the nodes in the fetched rounds should be processed to ensure prompt ordering. Otherwise the ordering can be arbitrarily delayed until a round that mis-matches the parity of last ordered round is processed for ordering.

### Test Plan

Smoke tests pass

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #11977** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11977?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11977?utm_source=stack-comment).

**Review by sasha8 — COMMENTED** [review]
STAMP

**Review by sasha8 — APPROVED** [review]
STAMP

File: consensus/src/dag/dag_handler.rs [hunk 1]
```diff
@@ -160,6 +160,8 @@ impl NetworkHandler {
                             Ok(certified_node) => {
                                 if let Err(e) = dag_driver_clone.process(certified_node).await {
                                     warn!(error = ?e, "error processing certified node fetch notification");
+                                } else {
+                                    dag_driver_clone.try_order_all();
```
Comment (zekun000) [inline consensus/src/dag/dag_handler.rs:164]: we may want to call it even it's an error?

**Review by zekun000 — APPROVED** [review]

File: consensus/src/dag/dag_handler.rs (hunk 1 above)
Comment (ibalajiarun) [reply consensus/src/dag/dag_handler.rs:164]: discussed offline. If there is an error anywhere in the call, the node doesn't get into the dag store so don't need to be ordered.

**Pull request merged**